sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -830,6 +831,22 @@ class UDFSuite extends QueryTest with SharedSparkSession {
       Row(ArrayBuffer(100)))
   }
 
+  test("SPARK-46586: UDF should not fail on immutable.ArraySeq") {
+    val myUdf1 = udf((a: immutable.ArraySeq[Int]) =>
+      immutable.ArraySeq.unsafeWrapArray[Int](Array(a.head + 99)))
+    checkAnswer(Seq(Array(1))
+      .toDF("col")
+      .select(myUdf1(Column("col"))),
+    Row(ArrayBuffer(100)))
+
+     val myUdf2 = udf((a: immutable.ArraySeq[Int]) =>
+      immutable.ArraySeq.unsafeWrapArray[Int](a.appended(5).appended(6).toArray))
```

Comment:
super nit: isn't it more common to use `:+` to create new immutable collection with new elements?

Reply:
Okay, let me improve it.